src/lib/getIgnoredUpgradesDueToEnginesNode.ts

```diff
@@ -21,13 +21,13 @@ export async function getIgnoredUpgradesDueToEnginesNode(
   if (!options.nodeEngineVersion) return {}
   const optionsEnginesNodeMinVersion = minVersion(options.nodeEngineVersion)?.version
   if (!optionsEnginesNodeMinVersion) return {}
-  const [upgradedLatestVersions] = await upgradePackageDefinitions(current, {
+  const [upgradedLatestVersions, latestVersionResults] = await upgradePackageDefinitions(current, {
     ...options,
     enginesNode: false,
     nodeEngineVersion: undefined,
     loglevel: 'silent',
   })
-  const enginesNodes = await getEnginesNodeFromRegistry(upgradedLatestVersions, options)
+  const enginesNodes = await getEnginesNodeFromRegistry(latestVersionResults, options)
```

Comment:
Can you explain this change for me?

Reply:
upgradedLatestVersions contains specs (`^4.0.0`) which can't be used to get the info from npm registry. So we use the actual versions instead

Reply:
First, in that case, let's change the first param of `getEnginesNodeFromRegistry` to use `Version` instead of `VersionSpec`.

i.e.:

```ts
async function getEnginesNodeFromRegistry(packageMap: Index<Version>, options: Options)
```

Eventually we should change `Version` to a branded type so that the type system can enforce the distinction between an exact version and a version range.

---

Second, `latestVersionResults` is not filtered the way that `upgradedLatestVersions` is. Is that desired? I'm thinking we probably don't need to look up the engines node version for dependencies that have already been filtered out due to the `minimal` or `filterResults` options.

This filters down `latestVersionResults` to just the keys in `upgradedLatestVersions`:

```ts
const filteredLatestVersionResults = pickBy(latestVersionResults, (spec, dep) => upgradedLatestVersions[dep])
```

Reply:
I already changed getEnginesNodeFromRegistry to use Version.
Yes, branded types might be the way thought it comes with annoying boilerplate :-(
Yes you are correct, applied a fix